src/app/services/server-authentication.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import { ServerApiService } from './server-api.service';
+import { Observable } from 'rxjs';
+import { Response } from '../../../api-objects/GenericResponse'
+import { LoginSession } from 'api-objects/LoginSession';
+
+
+@Injectable()
+export class ServerAuthenticationService {
+    //constructor(private serverApiService: ServerApiService) { }
+    constructor(private serverApiService: ServerApiService) { }
+
+    public login(username: string, password: string): Observable<Response<LoginSession>> {
```

Comment:
Return observable VOID not the login session

Reply:
Currently working on how to map out the observable. I'd like to file this as a bug however _as it does not inhibit functionality._

Reply:
To change the type of `Observable`, you need to not use `tap`.